apps/vc-api/src/vc-api/credentials/dtos/verify-options.dto.ts

```diff
@@ -44,7 +44,7 @@ export class VerifyOptionsDto implements VerifyOptions {
     description:
       'A challenge provided by the requesting party of the proof. For example 6e62f66e-67de-11eb-b490-ef3eeefa55f2'
   })
-  challenge?: string;
+  challenge: string;
```

Comment:
I don't think we should remove the optional marker here. My reading of the spec is that the `challenge` property is optional https://w3c-ccg.github.io/vc-api/#verify-presentation 
Is there a particular reason it was made mandatory?
```suggestion
  challenge?: string;
```

Reply:
@jrhender In credo the `challenge` parameter is mandatory in the `W3cVerifyPresentationOptionsBase` interface.
The challenge can either be populated via the VPs proof object(here's its an optional field) or via the options object.
![image](https://github.com/user-attachments/assets/9bd52d85-6f2a-420b-910d-51978de874a6)

Reply:
```
const w3cVerifyPresentationOptions: W3cVerifyPresentationOptions = {
      presentation: this.transformToCredoPresentation(vp, W3cJsonLdVerifiablePresentation),
      challenge: options.challenge ?? vp.proof.challenge
    };
```
In our implementation we are setting the challenge either from the options or proof object.
`challenge` param is optional in `proof` object. And since `challenge` is mandatory for credo we removed it as an optional marker.

Another approach that i could think of is - 
```
challenge: vp.proof.challenge ?? options.challenge ?? random_challenge_generator();
```

Reply:
Thanks for the explanation and context @ShikharBhatt . Thinking about it some more, I think that we should indeed make the challenge mandatory in the verify `options` as you describe. I think a challenge should be used to prevent replay attacks and so I think it's reasonable to ask the verifier to explicitly confirm the challenge that they expect for this presentation. So I think we can keep the code as it is.